modules/ogc/net.opengis.csw/src/net/opengis/cat/csw20/impl/QueryTypeImpl.java

```diff
@@ -374,7 +375,7 @@ public boolean eIsSet(int featureID) {
             case Csw20Package.QUERY_TYPE__CONSTRAINT:
                 return constraint != null;
             case Csw20Package.QUERY_TYPE__SORT_BY:
-                return SORT_BY_EDEFAULT == null ? sortBy != null : !SORT_BY_EDEFAULT.equals(sortBy);
+                return SORT_BY_EDEFAULT == null ? sortBy != null : !Arrays.equals(SORT_BY_EDEFAULT, sortBy);
```

Comment:
This appears to be cleaning up a warning unrelated to the current activity. Much of this code is generated and thus hard suppress warnings.

If you wish to keep this change you need to set the javadoc `@generated` to `@generated false`, but it may be smarter to back out the change.

Reply:
Okay this change is a not as straight forward as the boolean boxing.   I can revert it.

Reply:
Has been reverted.

Reply:
Actually when I enable the error prone compiler in IntelliJ i am getting this error and can't compile the project.

`Error:(377, 92) java: [ArrayEquals] Reference equality used to compare arrays
    (see https://errorprone.info/bugpattern/ArrayEquals)
  Did you mean 'return SORT_BY_EDEFAULT == null ? sortBy != null : !Arrays.equals(SORT_BY_EDEFAULT, sortBy);'?`

It is my understanding that the error prone compiler is required by GeoTools and I can't compile the code the way it is in IntelliJ.

Reply:
I am never quite sure how to use error prone in IntelliJ, perhaps @aaime has guidance. Does error prone require some configuration to ignore these autogenerated folders?

Reply:
I don't use errorprone in IntelliJ, just run it in maven explicitly once I'm done with the basic development, and fix whatever pops up.

Reply:
I used the [IntelliJ IDEA (Official) plugin](https://plugins.jetbrains.com/plugin/7349-error-prone-compiler) and configured it in IntelliJ settings:

File | Settings | Build, Execution, Deployment | Compiler | Java Compiler

jetbrains://idea/settings?name=Build%2C+Execution%2C+Deployment--Compiler--Java+Compiler

GitHub doesn't seem to support their fancy application link to the settings.

Without it I could not compile to run the project, but with it would compile them so I could run tests in the IDE but until I fixed the "error prone" errors it wouldn't let me run tests and if i set the Run Configuration to "Build, no error check" in the Before launch section it would run old stale code.